src/Result.ts

```diff
@@ -81,3 +83,9 @@ type HasInlineFragment<T extends SelectionSet<any> | undefined> = T extends Sele
   : never;
 
 type Merge<M, N> = Omit<M, keyof N> & N;
+
+export type SelectionResult<TSelection> =
```

Comment:
Ah yeah I was planning on actually making this the default for our standard `Result` type. Thoughts on renaming our existing `Result` to `InnerResult` instead (so we can export this higher-level API as `Result`)?

Reply:
Sure, that sounds better. 

I wanted to avoid a breaking change, but I can update that if you prefer exposing as Result.

Reply:
True. Let's go with this for now.

Reply:
And as always thank you for your contribution!